### PR TITLE
(feat): add visual matcher

### PR DESCRIPTION
**Pull request by christian-bromann**
First draft of visual matchers. Needs some testing and documentation.

I decided for this interface:

```ts
it(`should match an element successful with a baseline for '${browserName}'`, async () => {
    await expect($('.hero__title-logo')).toMatchElementSnapshot('wdioLogo', 0, {
        removeElements: [await $('nav.navbar')]
    })
})

it(`should match a full page screenshot successful with a baseline for '${browserName}'`, async () => {
    await expect(browser).toMatchFullPageSnapshot('fullPage', 0, {
        fullPageScrollTimeout: 1500,
        hideAfterFirstScroll: [
            await $('nav.navbar'),
        ],
    })
})

it(`should match a tabbable screenshot successful with a baseline for '${browserName}'`, async () => {
    await expect(browser).toMatchTabbablePageSnapshot('tabbable', 0, {
        hideAfterFirstScroll: [
            await $('nav.navbar'),
        ]
    })
})
```

implements #140

### CODE REVIEW

**Comment by christian-bromann** [comment]
@wswebcreation let me know what you think!

**Comment by christian-bromann** [comment]
Add docs for them in https://github.com/webdriverio/expect-webdriverio/pull/1409

**Comment by wswebcreation** [comment]
> Add docs for them in [webdriverio/expect-webdriverio#1409](https://github.com/webdriverio/expect-webdriverio/pull/1409)

Done this one, added comments to it

**Review by wswebcreation — APPROVED** [review]
LGTM, thanks!

**Comment by christian-bromann** [comment]
> What do you think?

Great idea, will add the feature.

**Comment by christian-bromann** [comment]
Improved the matcher failing message:

```
 "spec" Reporter:
------------------------------------------------------------------
[Safari 16.6 macOS #0-0] Running: Safari (v16.6) on macOS
[Safari 16.6 macOS #0-0] Session ID: 14b39b5f154444faa0970b43c6e36565
[Safari 16.6 macOS #0-0]
[Safari 16.6 macOS #0-0] » /tests/specs/desktop.spec.ts
[Safari 16.6 macOS #0-0] @wdio/visual-service desktop
[Safari 16.6 macOS #0-0]    ✖ should compare an element successful with a baseline for 'safari-16'
[Safari 16.6 macOS #0-0]    ✓ should compare a full page screenshot successful with a baseline for 'safari-16'
[Safari 16.6 macOS #0-0]    ✓ should compare a tabbable screenshot successful with a baseline for 'safari-16'
[Safari 16.6 macOS #0-0]
[Safari 16.6 macOS #0-0] 2 passing (56.5s)
[Safari 16.6 macOS #0-0] 1 failing
[Safari 16.6 macOS #0-0]
[Safari 16.6 macOS #0-0] 1) @wdio/visual-service desktop should compare an element successful with a baseline for 'safari-16'
[Safari 16.6 macOS #0-0] Expected image to have a mismatch percentage of 0%, but was 46.878%

Please compare the images manually and update the baseline if the new screenshot is correct.
Baseline: /Users/christian.bromann/Sites/WebdriverIO/projects/visual-testing/tests/sauceLabsBaseline/desktop_safari/wdioLogo-macOS12-16-1366x768.png
Actual Screenshot: /Users/christian.bromann/Sites/WebdriverIO/projects/visual-testing/.tmp/actual/desktop_safari/wdioLogo-macOS12-16-1366x768.png
Difference: /Users/christian.bromann/Sites/WebdriverIO/projects/visual-testing/.tmp/diff/desktop_safari/wdioLogo-macOS12-16-1366x768.png

See https://webdriver.io/docs/api/visual-regression.html for more information.
```

**Comment by christian-bromann** [comment]
Assertions can now be written without an expected result, it defaults to `0`:
```ts
await expect($('.hero__title-logo')).toMatchElementSnapshot('wdioLogo', {
    removeElements: [await $('nav.navbar')]
})
```

**Comment by christian-bromann** [comment]
Let me know what you all think!

File: packages/service/src/matcher.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+     * make sure `options` is an object
+     */
+    if (typeof options !== 'object') {
+        options = {}
+    }
+
+    /**
+     * overwrite `returnAllCompareData` to allow us to provide a better assertion message
+     */
+    options.returnAllCompareData = true
```
Comment (wswebcreation) [inline packages/service/src/matcher.ts:94]: 👍

File: packages/service/src/matcher.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+     * expected value is a number
+     *
+     * ```ts
+     * expect(browser).toMatchScreenSnapshot('foo', 0)
+     * ```
+     */
+    if (typeof expected === 'number') {
+        return {
+            pass: result.misMatchPercentage === expected,
+            message: () => (
```
Comment (wswebcreation) [inline packages/service/src/matcher.ts:47]: ❤️

**Pull request merged**